File: frontend/src/types/address.ts
// types/address.ts

/**
 * Khớp hoàn toàn với OwnerType.java
 */
export type OwnerType = 'TENANT' | 'USER'; 

export interface AddressResponseDTO {
  id: number;
  fullAddress: string;
  isDefault: boolean;
}

/**
 * Khớp hoàn toàn với AddressDetailResponseDTO.java
 */
export interface AddressDetailResponseDTO extends AddressResponseDTO {
  tenantId: number;
  ownerType: OwnerType;
  ownerId: number;

  houseNumber: string | null; // Java cho phép null
  street: string;             // Backend @NotBlank nên không để dấu ?
  ward: string;               // Backend @NotBlank
  district: string | null;
  province: string | null;
  country: string | null;
}

/**
 * Khớp với AddressRequestDTO.java (Dùng cho cả Create và Update)
 */
export interface AddressRequestDTO {
  ownerType: OwnerType; // @NotNull
  ownerId: number;      // @NotNull
  
  street: string;       // @NotBlank
  houseNumber?: string;
  ward: string;         // @NotBlank
  district?: string;
  province?: string;
  country?: string;
  isDefault: boolean;
}

// Nếu bạn muốn tách riêng Update để có ID (theo logic API PUT thông thường)
export interface UpdateAddressRequest extends AddressRequestDTO {
  id: number;
}

File: backend/src/main/java/com/chatbot/shared/address/model/Address.java
package com.chatbot.shared.address.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Filter;

@Entity
@Table(
    name = "addresses",
    indexes = {
        @Index(name = "idx_address_tenant", columnList = "tenant_id"),
        @Index(name = "idx_address_owner", columnList = "tenant_id, owner_type, owner_id")
    }
)
@Filter(name = "tenantFilter", condition = "tenant_id = :tenantId")
// ---------------------
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Tenant ID for multi-tenant architecture
     */
    @Column(name = "tenant_id")
    private Long tenantId;

    /**
     * Loại owner: USER, END_USER, ORDER, SHOP...
     */
    @Enumerated(EnumType.STRING)
    @Column(name = "owner_type", nullable = false, length = 30)
    private OwnerType ownerType;

    /**
     * ID của owner tương ứng với ownerType
     */
    @Column(name = "owner_id", nullable = false)
    private Long ownerId;

    // ===== Địa chỉ =====
    @Column(name = "house_number")
    private String houseNumber;

    @Column(length = 255)
    private String street;

    @Column(length = 255)
    private String ward;

    @Column(length = 255)
    private String district;

    @Column(length = 255)
    private String province;

    @Column(length = 255)
    private String country;
}